sale_blanket_order/models/blanket_orders.py

```diff
@@ -0,0 +1,329 @@
+# coding: utf-8
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models, api, _
+from odoo.exceptions import Warning
+from odoo.tools import float_is_zero
+
+
+class BlanketOrder(models.Model):
+    _name = 'sale.blanket.order'
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+    _description = 'Blanket Order'
+
+    @api.model
+    def _get_default_team(self):
+        return self.env['crm.team']._get_default_team_id()
+
+    @api.model
+    def _default_currency(self):
+        return self.env.user.company_id.currency_id
+
+    @api.model
+    def _default_company(self):
+        return self.env.user.company_id
+
+    name = fields.Char(
+        default='Draft',
+        readonly=True
+    )
+    partner_id = fields.Many2one(
+        'res.partner', string='Partner', readonly=True,
+        states={'draft': [('readonly', False)]})
+    lines_ids = fields.One2many(
+        'sale.blanket.order.line', 'order_id', string='Order lines',
+        copy=True)
+    pricelist_id = fields.Many2one(
+        'product.pricelist', string='Pricelist', required=True, readonly=True,
+        states={'draft': [('readonly', False)]})
+    currency_id = fields.Many2one(
+        'res.currency', related='pricelist_id.currency_id', readonly=True)
+    payment_term_id = fields.Many2one(
+        'account.payment.term', string='Payment Terms', readonly=True,
+        states={'draft': [('readonly', False)]})
+    confirmed = fields.Boolean(default=False)
```

Comment:
That's default

Reply:
"Explicit is better than implicit" :smile: